boot/nuttx/main.c

```diff
@@ -32,6 +32,27 @@
 
 #include "flash_map_backend/flash_map_backend.h"
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Should we perform board-specific driver initialization?  There are two
+ * ways that board initialization can occur:  1) automatically via
+ * board_late_initialize() during bootupif CONFIG_BOARD_LATE_INITIALIZE
+ * or 2).
+ * via a call to boardctl() if the interface is enabled
+ * (CONFIG_BOARDCTL=y).
+ * If this task is running as an NSH built-in application, then that
+ * initialization has probably already been performed otherwise we do it
+ * here.
+ */
+
+#undef NEED_BOARDINIT
+
+#if defined(CONFIG_BOARDCTL) && !defined(CONFIG_NSH_ARCHINIT)
+#  define NEED_BOARDINIT 1
+#endif
+
```

Comment:
```suggestion
```
This entire block is not used anywhere and should be removed.
`CONFIG_BOARDCTL` will be always defined to **1** since `BOARDCTL` is a dependency for the MCUboot bootloader.

Reply:
Solved in [3bcacea](https://github.com/mcu-tools/mcuboot/pull/1455/commits/3bcacea29b66df0c1d26cc456fbdbbd5c525fae0). I have not remove the `CONFIG_BOARDCTL` check, even if it is always **1**, to homogenize with other units the way of initializing board drivers.

For example: 
[lvgdemo](https://github.com/apache/incubator-nuttx-apps/blob/master/examples/lvgldemo/lvgldemo.c)

Reply:
Fine.